src/user.rs

```diff
@@ -195,13 +219,25 @@ impl User {
                     // it's time to log out
                     self.log_out(cancel_sync.clone()).await;
                 } else {
-                    match pick_random_action(context.config.simulation.probability_to_act) {
+                    match pick_random_action(
+                        context.config.simulation.probability_to_act,
+                        context.config.simulation.channels_load,
+                    ) {
                         SocialAction::SendMessage => {
-                            self.send_message(pick_random_room(rooms).await).await
+                            self.send_message(pick_random_room(direct_messages).await)
+                                .await
                         }
                         SocialAction::AddFriend => self.add_friend(context).await,
                         SocialAction::LogOut => self.log_out(cancel_sync.clone()).await,
                         SocialAction::UpdateStatus => self.update_status().await,
+                        SocialAction::CreateChannel => {
+                            if channels.read().await.len()
+                                < context.config.simulation.channels_per_user
+                                && context.config.simulation.channels_load
```

Comment:
Let's move this if to `create_channel` function and add debug logs for users that reach the limit of channels.